### DISCUSSION TITLE
Tag support is now missing as of v5?

### DISCUSSION THREAD

**Discussion by Braunson** [spatie/laravel-newsletter#304] [Q&A] [answered]
Looks like v4 had addTags/removeTags but as of v5, they've been removed? Any reason as-to why?

https://github.com/spatie/laravel-newsletter/blob/4.11.0/src/Newsletter.php#L154-L185

**Comment by titizali** [comment]
Yes, I had to extend the MailChimpDriver.php to make it work again. 

Update: You can use the subscribeOrUpdate method and pass ['tags'=>['tag1','tag2']] into it. For removing tags, you can clear tags with ['tags'=>[]] and add all the relevant tags back in afterwards.

**Reply by Braunson** [reply, marked as answer]
Thanks, I just ended up extending to add back addTags/removeTags methods

**Reply by titizali** [reply]
Were you able to do that for MailCoach? The API seems a little different

**Reply by Braunson** [reply]
Sorry for this project I wasn't using MailCoach.

**Comment by titizali** [comment]
I just used the built in tag feature to rewrite the tags every time I edit
the subscriber. Kinda sucks as a solution and prone to errors. I'm
integrating with Mailcoach.

You can add the AddTag function back from the previous version if you're
using Mailchimp. It doesn't work with Mailcoach.
